#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_infinite_loop_r5264.dpatch by Daniel Leidert (dale) <daniel.leidert@wgdd.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: * keygen.c (ask_user_id): Avoid infinite loop in case of invalid
## DP:   data.
## DP:
## DP: <URL:https://bugs.g10code.com/gnupg/issue1186>
## DP: <URL:http://lists.gnupg.org/pipermail/gnupg-commits/2010-February/008776.html>

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' trunk~/g10/keygen.c trunk/g10/keygen.c
--- trunk~/g10/keygen.c	2009-07-29 10:42:59.000000000 +0200
+++ trunk/g10/keygen.c	2010-03-21 15:14:33.547740573 +0100
@@ -1927,7 +1927,7 @@
 	    if( strlen(ansstr) != 10 )
 		BUG();
 	    if( cpr_enabled() ) {
-		answer = xstrdup(ansstr+6);
+                answer = xstrdup (ansstr + (fail?8:6));
 		answer[1] = 0;
 	    }
 	    else {
